libraries/pbrlib/genosl/mx_dielectric_bsdf.osl

```diff
@@ -1,36 +1,15 @@
-#include "lib/mx_microfacet_specular.osl"
-
 void mx_dielectric_bsdf(float weight, color tint, float ior, vector2 roughness, normal N, vector U, string distribution, string scatter_mode, output BSDF bsdf)
 {
-    if (scatter_mode == "T")
+    if (scatter_mode == "R")
     {
-        bsdf.response = tint * weight * microfacet(distribution, N, U, roughness.x, roughness.y, ior, 1);
-        bsdf.throughput = tint * weight;
-        return;
+        bsdf = weight * dielectric_bsdf(N, U, tint, color(0.0), roughness.x, roughness.y, ior, distribution);
     }
-
-    float NdotV = clamp(dot(N,-I), M_FLOAT_EPS, 1.0);
-    float F0 = mx_ior_to_f0(ior);
-    float F = mx_fresnel_schlick(NdotV, F0);
-
-    // Calculate compensation for multiple scattering.
-    // This should normally be done inside the closure
-    // but since vanilla OSL doesen't support this we
-    // add it here in shader code instead.
-    vector2 safeAlpha = clamp(roughness, M_FLOAT_EPS, 1.0);
-    float avgAlpha = mx_average_alpha(safeAlpha);
-    float comp = mx_ggx_energy_compensation(NdotV, avgAlpha, F);
-
-    // Calculate throughput from directional albedo.
-    float dirAlbedo = mx_ggx_dir_albedo(NdotV, avgAlpha, ior) * comp;
-    bsdf.throughput = 1.0 - dirAlbedo * weight;
-
-    if (scatter_mode == "R")
+    else if (scatter_mode == "T")
     {
-        bsdf.response = tint * weight * comp * microfacet(distribution, N, U, safeAlpha.x, safeAlpha.y, ior, 0);
+        bsdf = weight * dielectric_bsdf(N, U, color(0.0), tint, roughness.x, roughness.y, ior, distribution);
     }
     else
     {
-        bsdf.response = tint * weight * comp * microfacet(distribution, N, U, safeAlpha.x, safeAlpha.y, ior, 2);
+        bsdf = weight * dielectric_bsdf(N, U, tint, tint, roughness.x, roughness.y, ior, distribution);
```

Comment:
For refraction to occur this has to be paired with a medium closure. Is it expected that the user will have a medium node in the graph if they want refraction?

Reply:
If refraction with absorption/scattering is requested then it is expected that the user sets a `medium_vdf` below the `dielectric_bsdf`, using the `layer` operator, in the MaterialX graph. 

But it would be good if the `dielectric_bsdf` could support basic refraction without having the `medium_vdf` present. In that case it would use the same ior for both the reflection and transmission parts. This would ensure backwards compatibility as this is the current behavior before we add the `medium_vdf` to MaterialX, and it's the behavior used in MDL for example.

I'm imagining the following behavior could be used for the two dielectric BSDF's.

- `dielectric_bsdf` If transmission is enabled but no medium is present use the one and only ior for both reflection and transmission.
- `generalized_schlick_bsdf` If transmission is enabled but no medium is present use ior=1.0 for transmission and hence get straight transmission without any refraction.

Reply:
To my mind it would be better to just have ior=1.0 and get no refraction for `generalized_schlick_bsdf` if no medium is present. But this is mainly because I find it so unintuitive that f0 will control how light is refracted. It seems very hard to control for an artist and especially since refraction has such a drastic effect on the look. But this is my personal preference and I'm open to other suggestions.

Reply:
Perhaps we should start with the current MaterialX behavior, where IOR is derived from F0 in `generalized_schlick_bsdf`, but discuss this further in the ASWF Slack channel for this topic.  Ideally, I'd like to bring Naty Hoffman and Andre Mazzone from Lucasfilm/ILM into that conversation, since they've given that question additional thought in a production context.

Reply:
Sounds good. Thanks for the input @jstone-lucasfilm.